File: Javascript/ToBeOrNotToBe.js
/**
 * Write a function expect that helps developers test their code. It should take in any value val and return an object with the following two functions.
 * toBe(val) accepts another value and returns true if the two values === each other. If they are not equal, it should throw an error "Not Equal".
 * notToBe(val) accepts another value and returns true if the two values !== each other. If they are equal, it should throw an error "Equal".
 * expect(5).toBe(5); // true
 * expect(5).notToBe(5); // throws "Equal"
 */
var expect = function(value) {
    return {
        toBe : (param) => {
            console.log(value + " tobe" + " "+ param)
        },
        notToBe : (param) => {
            console.log(value + " notToBe")
        }
    }    
};


console.log(expect("hi").toBe(5));

File: python/ToBeOrNotToBe.py
class Expect:
    def __init__(self, value):
        self.value = value

    def toBe(self, param):
        if self.value == param:
            return True
        else:
            raise ValueError("Not Equal")

    def notToBe(self, param):
        if self.value != param:
            return True
        else:
            raise ValueError("Equal")

def expect(value):
    return Expect(value)

# Example Usage:
try:
    print(expect(5).toBe(5))  # Output: True
    print(expect(5).toBe(6))  # Raises ValueError: Not Equal
except ValueError as e:
    print(e)  # Output: Not Equal

try:
    print(expect(5).notToBe(6)) # Output: True
    print(expect(5).notToBe(5)) # Raises ValueError: Equal
except ValueError as e:
    print(e) # Output: Equal